crates/chain-config/src/config/codec.rs

```diff
@@ -0,0 +1,151 @@
+mod decoder;
+mod encoder;
+mod parquet;
+
+pub use decoder::{
+    Decoder,
+    IntoIter,
+};
+pub use encoder::Encoder;
+
+use std::fmt::Debug;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Group<T> {
+    pub index: usize,
+    pub data: Vec<T>,
+}
+type GroupResult<T> = anyhow::Result<Group<T>>;
+
+#[cfg(test)]
+mod tests {
+    use std::ops::Range;
+
+    use crate::{
+        config::{
+            contract_balance::ContractBalance,
+            contract_state::ContractStateConfig,
+        },
+        CoinConfig,
+        ContractConfig,
+        MessageConfig,
+    };
+
+    use itertools::Itertools;
+
+    use super::*;
+
+    #[test]
+    fn writes_then_reads_written() {
+        let group_size = 100;
+        let num_groups = 10;
+        let starting_group_index = 3;
+        {
```

Comment:
These could be separate tests for `json` and `parquet`. Easier to read.

Nit: maybe add `roundtrip` to the name too. I feel like that's the _expected_ name for tests like this--maybe not.